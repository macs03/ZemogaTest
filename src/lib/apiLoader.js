export default class ApiLoader {
  static boot(App) {
    const baseUrl = 'https://jsonplaceholder.typicode.com/';

    App.api = {
      get: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'GET',
        });
      },
      post: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'POST',
        });
      },
      delete: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'DELETE',
        });
      },
    };
  }
}
