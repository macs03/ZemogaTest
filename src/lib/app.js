import AssetsLoader from './assetsLoader';
import ApiLoader from './apiLoader';

const globalLoaders = [AssetsLoader, ApiLoader];

export default class App {
  static api = undefined;

  static boot() {
    return Promise.all(globalLoaders.map((s) => s.boot(App)))
      .then(() => {
        console.info('App is ready!!!');
      })
      .catch((err) => console.error('Error while booting the app', err));
  }
}
