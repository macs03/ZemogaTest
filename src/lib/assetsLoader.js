import {Assets} from 'react-native-ui-lib';

export default class AssetsLoader {
  static boot() {
    Assets.loadAssetsGroup('icons', {
      refresh: require('../assets/icons/refresh.png'),
      favorite: require('../assets/icons/star.png'),
      trash: require('../assets/icons/delete.png'),
      chevron: require('../assets/icons/chevronRight.png'),
    });
  }
}
