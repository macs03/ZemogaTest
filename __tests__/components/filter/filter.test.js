/**
 * @format
 */

import 'react-native';
import React from 'react';
import Filter from '../../../src/components/filter/filter';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {on, Interactions, findByTestId} from '../../../src/lib/user-simulator';

it('renders correctly', () => {
  const tree = renderer.create(<Filter onFilter={() => {}} />);
  expect(tree).toMatchSnapshot();
});

it('calls filtering after tap button', () => {
  const spy = jest.fn();
  const tree = renderer.create(<Filter onFilter={spy} />);

  const button = findByTestId(tree, 'filtering-1');

  on(button).simulate(Interactions.PRESS);

  expect(spy).toHaveBeenCalledWith(1);
});
