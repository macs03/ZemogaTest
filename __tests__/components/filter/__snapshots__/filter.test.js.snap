// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#16AA16",
        "borderColor": "#16AA16",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": "60%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "50%",
      }
    }
    testID="filtering-0"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#16AA16",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      All
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#16AA16",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": "60%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "50%",
      }
    }
    testID="filtering-1"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#16AA16",
          },
          false,
        ]
      }
    >
      Favorite
    </Text>
  </View>
</View>
`;
